docs/DeveloperGuide.md

```diff
@@ -4,10 +4,150 @@
+
+### Add Command Feature
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
+A key functionality of MindMyMoney is the ability to add and track user expenditure. A user can add in a new expenditure
+by specifying the payment type, the category, the description of the item, the cost of the item and the date and the
+month purchased.
+
+1. The `Parser` component parses user input and returns the new `AddCommand` object to the
+   `MindMyMoney`.
+2. `AddCommand` instantiates `addInput`, `expenditureList` and `creditCardList`.
+3. The application invokes `Addcommand.execute` to execute user instruction.
+4. During the execution, `Addcommand.execute` will parse through user input to obtain the `EXPENDITURE`, `CATEGORY`,
+   `DESCRIPTION`, `AMOUNT` and `TIME` fields.
+5. The `Addcommand` object instantiates a new `Expenditure` object with the aforementioned 5 fields and adds them
+   into the `ExpenditureList`.
+6. The `Addcommand` object prints a list to show the user what it has saved.
+
+#### Design considerations
+Aspect: How to ask user for the 5 fields of input
+* Alternative 1 (current choice): User is asked to put in all 5 fields at once, separated using flags.
+    * Pros: Faster input, user can enter an expenditure using a single input.
+    * Cons: User must be able to remember all the flags and its sequence.
+
+* Alternative 2: User is asked iteratively to put in all 5 fields, prompted by a message after each input.
+    * Pros: Beginner friendly, easily understandable, no need to remember flags.
+    * Cons: Slower, implementation when user is familiar with the application.
+
+### Calculate Command feature
+
+To enable users to view their finances in a more meaningfully, MindMyMoney does calculations to present financial data
+that is actionable for the users.
+
+![calculate_command_sequence_diagram](images/gif_loading.gif)
+
+### List Command feature
+To enable users to view their current list of added expenditures. 
+
+#### Design considerations
+Aspect: How to make the list command easily tested using JUnit testing
+* Alternative 1 (current choice): Abstract the conversion of Expenditure to String in a separate listToString() method
+  * Pros: Easily tested using JUnit testing, by checking the String that the listToString() method returns
+  * Cons: Added layer of abstraction that may be deemed redundant
+
+* Alternative 2: Print directly in the ListCommand.executeCommand() method 
+  * Pros: Easily implemented with lesser lines of code as the code is minimalist
+  * Cons: JUnit testing would require I/O redirection prior to checking the output matches expectations
+
+![list_command_sequence_diagram](images/List_Command_Sequence_Diagram.png)
 
 ## Product scope
 ### Target user profile
```

Comment:
There is no information in the product scope, non-functional requirement, glossary and instruction for manual testing sections.
<img width="342" alt="image" src="https://user-images.githubusercontent.com/69478335/160959268-d33cad62-8ba2-4113-a03a-93673cfbd242.png">
<img width="607" alt="image" src="https://user-images.githubusercontent.com/69478335/160959293-b5601eaf-c6c1-4c16-9b65-f23427665b81.png">